other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java

```diff
@@ -1,76 +1,111 @@
+
+    private static final String SUBSCRIPTION_KEY = "<YOUR-SUBSCRIPTION-KEY>";
+
+    private static final String END_POINT = "<YOUR-ANOMALY-DETECTOR-END-POINT>";
+
+    private static final HttpPipelinePolicy AUTH_POLICY =
+            new AzureKeyCredentialPolicy(OCP_APIM_SUBSCRIPTION_KEY,
+                    new AzureKeyCredential(SUBSCRIPTION_KEY));
+
+    private static final AddHeadersPolicy ADD_HEADERS_POLICY =
+            new AddHeadersPolicy(new HttpHeaders().put(CONTENT_TYPE, APPLICATION_JSON));
+
+    private static final HttpPipeline HTTP_PIPELINE =
+            new HttpPipelineBuilder().policies(AUTH_POLICY, ADD_HEADERS_POLICY).build();
+
+    public static void main(String[] args) throws IOException {
+
+        AnomalyDetectorAsyncClient asyncClient = new AnomalyDetectorClientBuilder()
+                .endpoint(END_POINT).pipeline(HTTP_PIPELINE).buildAsyncClient();
+
+        JSONObject requestJson = new JSONObject(new JSONTokener((Main.class.getResourceAsStream("/request-data.json"))));
+        List<TimeSeriesPoint> timeSeriesPointList = new ArrayList<>();
+        JSONArray seriesArray = requestJson.getJSONArray("series");
+        for (int i = 0; i < seriesArray.length(); i++) {
+            JSONObject seriesObject = seriesArray.getJSONObject(i);
+            TimeSeriesPoint timeSeriesPoint = new TimeSeriesPoint()
+                    .setTimestamp(OffsetDateTime.ofInstant(Instant.parse(seriesObject.getString("timestamp")), ZoneOffset.UTC))
+                    .setValue(seriesObject.getFloat("value"));
+            timeSeriesPointList.add(timeSeriesPoint);
         }
-        JSONTokener tokener = new JSONTokener(is);
-        String content = new JSONObject(tokener).toString();
-        CloseableHttpClient client = HttpClients.createDefault();
-        HttpPost request = new HttpPost(uriBase);
-
-        // Request headers.
-        request.setHeader("Content-Type", "application/json");
-        request.setHeader("Ocp-Apim-Subscription-Key", subscriptionKey);
-
-        try {
-            StringEntity params = new StringEntity(content);
-            request.setEntity(params);
-
-            CloseableHttpResponse response = client.execute(request);
-            try {
-                HttpEntity respEntity = response.getEntity();
-                if (respEntity != null) {
-                    System.out.println("----------");
-                    System.out.println(response.getStatusLine());
-                    System.out.println("Response content is :\n");
-                    System.out.println(EntityUtils.toString(respEntity, "utf-8"));
-                    System.out.println("----------");
+
+        DetectRequest request = new DetectRequest()
+                .setGranularity(TimeGranularity.fromString(requestJson.getString("granularity")))
+                .setSeries(timeSeriesPointList);
+
+        asyncClient.detectLastPointWithResponse(request).subscribe(response -> {
```

Comment:
Is there a reason you want access to the HttpResponse object? We expect people to generally not care about that and instead, use `detectLastPoint`

Reply:
According to the comment, invoke method changed to [ detectLastPoint ].